File: Data Structures & Algorithms/maximum-subarray/submission-2.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        currsum = nums[0]
        maxsum = nums[0]
        n = len(nums)

        for i in range(1, n):
            currsum = max(nums[i], currsum + nums[i])
            maxsum = max(maxsum, currsum)
        
        return maxsum

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int j = 0;
        for(int i =1;i<nums.size();i++){
            while(j< nums.size() && i<nums.size()){
                if(nums[i] != nums[j]){
                    nums[j+1] = nums[i];
                    j++;
                }
                i++;
            }
        }
        return j+1;
    }
}; 